### PR TITLE
Fixing issues related api Product

### PR DESCRIPTION
**Pull request by HiranyaKavishani**
This PR will, 
1. Fix the issues related to runtime configuration in APIProduct ( Security, Response Caching, schema validation)
2. Fix the typo issue in API Product create
3. Hide the sample api product deployment
4. Fix the issue in validation

Related issue : https://github.com/wso2/product-apim/issues/6510

### CODE REVIEW

File: components/apimgt/org.wso2.carbon.apimgt.api/src/main/java/org/wso2/carbon/apimgt/api/model/APIProduct.java [hunk 1]
```diff
@@ -344,10 +345,32 @@ public String getAccessControlRoles() {
...
+    /**
+     * To set the JSON schema validation enable/disable.
+     *
+     * @param enableSchemaValidation Given Status.
+     */
+    public void setEnableSchemaValidation(boolean enableSchemaValidation) {
+        this.enableSchemaValidation = enableSchemaValidation;
+    }
+
+
```
Comment (bhathiya) [inline components/apimgt/org.wso2.carbon.apimgt.api/src/main/java/org/wso2/carbon/apimgt/api/model/APIProduct.java]: plz remove all extra new lines.

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/utils/APIUtil.java [hunk 2]
```diff
@@ -8975,6 +8988,17 @@ public static APIProduct getAPIProduct(GovernanceArtifact artifact, Registry reg
...
             apiProduct.setContextTemplate(artifact.getAttribute(APIConstants.API_OVERVIEW_CONTEXT_TEMPLATE));
+            apiProduct.setEnableSchemaValidation(Boolean.parseBoolean(artifact.getAttribute(APIConstants.
+                    API_OVERVIEW_ENABLE_JSON_SCHEMA)));
+            apiProduct.setResponseCache(artifact.getAttribute(APIConstants.API_OVERVIEW_RESPONSE_CACHING));
+
+            int cacheTimeout = APIConstants.API_RESPONSE_CACHE_TIMEOUT;
+            try {
+                cacheTimeout = Integer.parseInt(artifact.getAttribute(APIConstants.API_OVERVIEW_CACHE_TIMEOUT));
+            } catch (NumberFormatException e) {
+                //ignore
```
Comment (bhathiya) [inline components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/utils/APIUtil.java]: let's put a debug log.

File: components/apimgt/org.wso2.carbon.apimgt.api/src/main/java/org/wso2/carbon/apimgt/api/model/APIProduct.java [hunk 3]
```diff
@@ -344,10 +345,32 @@ public String getAccessControlRoles() {
...
     public String getApiSecurity() {
         return apiSecurity;
     }
 
+
+
+
```
Comment (bhathiya) [inline components/apimgt/org.wso2.carbon.apimgt.api/src/main/java/org/wso2/carbon/apimgt/api/model/APIProduct.java]: same as above

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIProviderImpl.java [hunk 4]
```diff
@@ -2001,8 +2001,8 @@ private void sendEmailNotification(API api) throws APIManagementException {
         }
     }
 
-    private void validateAndSetTransports(API api) throws APIManagementException {
-        String transports = api.getTransports();
+    private void validateAndSetTransports(API api, APIProduct apiProduct) throws APIManagementException {
```
Comment (uvindra) [inline components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIProviderImpl.java]: Its not a good design to have a single method like this to handle both API and APIProduct scenario, this leads to lots of null checks that can be avoided and it means the function user has to purposefully pass null as one of the parameters.

Lets define an overloaded function for APIProduct _void validateAndSetTransports(APIProduct apProduct)_ to handle this.

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIProviderImpl.java [hunk 5]
```diff
@@ -2022,10 +2026,11 @@ private void validateAndSetTransports(API api) throws APIManagementException {
      *
      * @param api Relevant API that need to be validated.
      */
-    private void validateAndSetAPISecurity(API api) {
+    private void validateAndSetAPISecurity(API api, APIProduct apiProduct) {
```
Comment (uvindra) [inline components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIProviderImpl.java]: As suggested previously lets add a separate overloaded function for APIProduct

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIProviderImpl.java [hunk 6]
```diff
@@ -2092,14 +2097,27 @@ private void validateAndSetAPISecurity(API api) {
...
-        api.setApiSecurity(apiSecurity);
+
+        String apiType;
+        if (apiProduct == null) {
+            apiType = "API";
+            api.setApiSecurity(apiSecurity);
+        } else {
+            apiType = "APIProduct";
+            apiProduct.setApiSecurity(apiSecurity);
+        }
```
Comment (uvindra) [inline components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIProviderImpl.java]: Please use the constant that is already defined to set the apiType for consistency

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIProviderImpl.java (hunk 5 above)
Comment (bhathiya) [reply components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIProviderImpl.java]: +1

File: components/apimgt/org.wso2.carbon.apimgt.api/src/main/java/org/wso2/carbon/apimgt/api/model/APIProduct.java (hunk 1 above)
Comment (HiranyaKavishani) [reply components/apimgt/org.wso2.carbon.apimgt.api/src/main/java/org/wso2/carbon/apimgt/api/model/APIProduct.java]: Fixed by 329d7041be4909124176d5d64c22a650a4979bf0

File: components/apimgt/org.wso2.carbon.apimgt.api/src/main/java/org/wso2/carbon/apimgt/api/model/APIProduct.java (hunk 3 above)
Comment (HiranyaKavishani) [reply components/apimgt/org.wso2.carbon.apimgt.api/src/main/java/org/wso2/carbon/apimgt/api/model/APIProduct.java]: Fixed by 329d7041be4909124176d5d64c22a650a4979bf0

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/utils/APIUtil.java (hunk 2 above)
Comment (HiranyaKavishani) [reply components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/utils/APIUtil.java]: Fixed in 310f49527926ae312930c648b63573a86c9f3030

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIProviderImpl.java (hunk 6 above)
Comment (HiranyaKavishani) [reply components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIProviderImpl.java]: Fixed in 32e0ed7cffd83e1e1bfb9eaf3eed84e0bb15a83a

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIProviderImpl.java (hunk 5 above)
Comment (HiranyaKavishani) [reply components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIProviderImpl.java]: Fixed in 32e0ed7cffd83e1e1bfb9eaf3eed84e0bb15a83a

**Pull request merged**